taichi/ui/common/scene_base.h

```diff
@@ -0,0 +1,63 @@
+#pragma once
+
+#include <vector>
+#include "taichi/ui/common/field_info.h"
+#include "taichi/ui/common/renderable_info.h"
+#include "taichi/ui/common/camera.h"
+#include "taichi/ui/utils/utils.h"
+
+TI_UI_NAMESPACE_BEGIN
+
+struct PointLight {
+  glm::vec3 pos;
+  glm::vec3 color;
+};
+
+struct MeshInfo {
+  RenderableInfo renderable_info;
+  glm::vec3 color;
+  float shininess;
+};
+
+struct ParticlesInfo {
+  RenderableInfo renderable_info;
+  glm::vec3 color;
+  float radius;
```

Comment:
Are we planning to do full spheres for these or billboard particles? Also it would be a nice touch to have RGBA colors for particles.

Reply:
Billboard particles. And for RGBA, IIUC we'll have to sort the particles? I’m not sure if we can do it fast enough. In PIC/FLIP simulations etc we can get to ~1e6 particles.